tap_trello/streams.py

```diff
@@ -76,59 +76,46 @@ def _get_window_state(self):
         window_state = {}
 
         start_date = self.config['start_date']
-        window_start_bookmark = (singer.get_bookmark(self.state, self.stream_id, 'window_start') or
-                                 singer.get_bookmark(self.state, self.stream_id, 'last_sync_end') or
-                                 start_date)
-        window_start = utils.strptime_to_utc(max(window_start_bookmark, start_date))
 
-        window_sub_end = singer.get_bookmark(self.state, self.stream_id, 'window_sub_end')
-        window_sub_end = window_sub_end and utils.strptime_to_utc(window_sub_end)
+        window_start_bookmark = (singer.get_bookmark(self.state, self.stream_id, 'window_start'))
+        window_start = utils.strptime_to_utc(max(window_start_bookmark, start_date))
 
-        next_window_start = singer.get_bookmark(self.state, self.stream_id, 'next_window_start')
-        next_window_start = next_window_start and utils.strptime_to_utc(next_window_start)
+        sub_window_end = singer.get_bookmark(self.state, self.stream_id, 'sub_window_end')
+        sub_window_end = sub_window_end and utils.strptime_to_utc(sub_window_end)
 
-        if window_sub_end is not None and next_window_start is None:
-            raise Exception("Invalid State: window_sub_end provided without a next_window_start.")
+        window_end = singer.get_bookmark(self.state, self.stream_id, 'window_end') # TODO: how should this consider end_date?
+        window_end = utils.strptime_to_utc(window_end)
 
-        return window_start, window_sub_end, next_window_start
+        return window_start, sub_window_end, window_end
 
-    def on_sync_started(self):
-        # Hook from `sync` implementer: Save when we started the overall sync, to use as a bookmark, later
-        now = utils.strftime(utils.now())
-        singer.write_bookmark(self.state, self.stream_id, "sync_end", min(self.config.get('end_date', now), now))
-        singer.write_bookmark(self.state, self.stream_id, "in_middle_of_sync", True)
+    def on_window_started(self):
+        if singer.get_bookmark(self.state, self.stream_id, 'sub_window_end') is None:
+            if singer.get_bookmark(self.state, self.stream_id, 'window_start') is None:
+                singer.write_bookmark(self.state, self.stream_id, "window_start", self.config['start_date'])
+            now = utils.strftime(utils.now())
+            singer.write_bookmark(self.state, self.stream_id, "window_end", min(self.config.get('end_date', now), now))
```

Comment:
This should have a guard around it, in case the tap gets interrupted in an edge case.

`if singer.get_bookmark(self.state, self.stream_id, 'window_end') is None:`